libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccount.m

```diff
@@ -274,13 +278,18 @@ - (void)setPhoto:(UIImage *)photo {
                     [SFSDKCoreLogger e:[self class] format:@"Unable to remove previous photo from disk: %@", error];
                 }
             }
-            NSData *data = UIImagePNGRepresentation(photo);
-            if (![data writeToFile:photoPath options:NSDataWritingAtomic error:&error]) {
-                [SFSDKCoreLogger e:[self class] format:@"Unable to write photo to disk: %@", error];
+            if (photo) {
+                NSData *data = UIImagePNGRepresentation(photo);
+                if (![data writeToFile:photoPath options:NSDataWritingAtomic error:&error]) {
+                    [SFSDKCoreLogger e:[self class] format:@"Unable to write photo to disk: %@", error];
+                }
             }
             [self willChangeValueForKey:@"photo"];
```

Comment:
We should move the willChange and didChange  inside the `if then` block?

Reply:
Which if then block are you thinking?